File: Day_27/Q107.java
import java.util.*;

//Class To Store Salary Details
class Employee{
    int id;
    String name;
    float basic_salary,bonus,total_salary;
}

class Q107{

    //Function To Input Salary Details
    static void input_salary(Employee e,Scanner sc){

        System.out.print("Enter Employee ID : ");
        e.id = sc.nextInt();

        sc.nextLine();

        System.out.print("Enter Employee Name : ");
        e.name = sc.nextLine();

        System.out.print("Enter Basic Salary : ");
        e.basic_salary = sc.nextFloat();

        System.out.print("Enter Bonus : ");
        e.bonus = sc.nextFloat();

        //Calculate Total Salary
        e.total_salary = e.basic_salary + e.bonus;
    }

    //Function To Display Salary Details
    static void display_salary(Employee e){

        System.out.println("\nEmployee ID : " + e.id);
        System.out.println("Employee Name : " + e.name);
        System.out.println("Basic Salary : " + e.basic_salary);
        System.out.println("Bonus : " + e.bonus);
        System.out.println("Total Salary : " + e.total_salary);
    }

    public static void main(String args[]){

        Scanner sc = new Scanner(System.in);

        int n;

        //Input Number Of Employees
        System.out.print("Enter Number Of Employees : ");
        n = sc.nextInt();

        //Array Of Objects
        Employee emp[] = new Employee[n];

        //Input Salary Records
        for(int i = 0;i < n;i++){

            emp[i] = new Employee();

            System.out.println("\nEnter Details Of Employee " + (i + 1));

            input_salary(emp[i],sc);
        }

        //Display Salary Records
        System.out.println("\n----- Salary Records -----");

        for(int i = 0;i < n;i++){
            display_salary(emp[i]);
        }

        sc.close();
    }
}

File: Day_27/Q107.cpp
#include<bits/stdc++.h>
using namespace std;

//Structure To Store Salary Details
struct Employee{
    int id;
    string name;
    float basic_salary,bonus,total_salary;
};

//Function To Input Salary Details
void input_salary(Employee &e){

    cout << "Enter Employee ID : ";
    cin >> e.id;

    cin.ignore();

    cout << "Enter Employee Name : ";
    getline(cin,e.name);

    cout << "Enter Basic Salary : ";
    cin >> e.basic_salary;

    cout << "Enter Bonus : ";
    cin >> e.bonus;

    //Calculate Total Salary
    e.total_salary = e.basic_salary + e.bonus;
}

//Function To Display Salary Details
void display_salary(Employee &e){

    cout << "\nEmployee ID : " << e.id << endl;
    cout << "Employee Name : " << e.name << endl;
    cout << "Basic Salary : " << e.basic_salary << endl;
    cout << "Bonus : " << e.bonus << endl;
    cout << "Total Salary : " << e.total_salary << endl;
}

int main(){

    int n;

    //Input Number Of Employees
    cout << "Enter Number Of Employees : ";
    cin >> n;

    //Vector Of Structure
    vector<Employee> emp(n);

    //Input Salary Records
    for(int i = 0;i < n;i++){

        cout << "\nEnter Details Of Employee " << i + 1 << endl;

        input_salary(emp[i]);
    }

    //Display Salary Records
    cout << "\n----- Salary Records -----\n";

    for(int i = 0;i < n;i++){
        display_salary(emp[i]);
    }

    return 0;
}